### PR TITLE
[12.0] [MIG] purchase_order_approval_block

### PR DESCRIPTION
**Pull request by Nikul-Chaudhary**
Purchase Order Approval Block
=============================

This module allows you to block the approval of an RFQ when an Approval Block Reason has been provided. Upon confirmation of an RFQ the orders will be waiting for approval by a Manager.

### CODE REVIEW

**Comment by Nikul-Chaudhary** [comment]
Depends on #609

**Comment by Nikul-Chaudhary** [comment]
@OCA/procurement-purchase-maintainers could you please review code ?

**Comment by Nikul-Chaudhary** [comment]
@pedrobaeza @rgarnau Could you please review it?

**Comment by tbaden** [comment]
could you please check the runbot status

**Comment by rousseldenis** [comment]
@Nikul-Chaudhary Have you asked original authors @jbeficent to change licensing from LGPL to AGPL ?

**Comment by Nikul-Chaudhary** [comment]
@rousseldenis In V11 used `AGPL`
Check please here: https://github.com/OCA/purchase-workflow/blob/11.0/purchase_order_approval_block/__manifest__.py#L2

**Comment by rousseldenis** [comment]
> @rousseldenis In V11 used `AGPL`
> Check please here: https://github.com/OCA/purchase-workflow/blob/11.0/purchase_order_approval_block/__manifest__.py#L2

Yes, but inconsistency with https://github.com/OCA/purchase-workflow/blob/11.0/purchase_order_approval_block/__manifest__.py#L21 and in other files.

I know @Eficent they do some modules with LGPL, that's why I asked.

**Comment by Nikul-Chaudhary** [comment]
> > @rousseldenis In V11 used `AGPL`
> > Check please here: https://github.com/OCA/purchase-workflow/blob/11.0/purchase_order_approval_block/__manifest__.py#L2
> 
> Yes, but inconsistency with https://github.com/OCA/purchase-workflow/blob/11.0/purchase_order_approval_block/__manifest__.py#L21 and in other files.
> 
> I know @Eficent they do some modules with LGPL, that's why I asked.

Ok Then Wait for @jbeficent review, if Good with `AGPL` then as it is. If not then Change as per answer.

**Comment by HviorForgeFlow** [comment]
@Nikul-Chaudhary let's keep LGPL-3 licence as it was set on previous versions.

**Comment by Nikul-Chaudhary** [comment]
> @Nikul-Chaudhary let's keep LGPL-3 licence as it was set on previous versions.

@hveficent Now Done, Can you please review it?

File: purchase_order_approval_block/models/purchase_approval_block_reason.py
Comment (jesusVMayor) [inline purchase_order_approval_block/models/purchase_approval_block_reason.py:8]: Add a _description to fix runbot warning.

File: purchase_order_approval_block/data/purchase_exception_data.xml
Comment (jesusVMayor) [inline purchase_order_approval_block/data/purchase_exception_data.xml]: https://github.com/OCA/server-tools/pull/1589

**Review by jesusVMayor — CHANGES_REQUESTED** [review]

**Comment by HviorForgeFlow** [comment]
> > @Nikul-Chaudhary let's keep LGPL-3 licence as it was set on previous versions.
> 
> @hveficent Now Done, Can you please review it?

Travis is :red_circle:

**Comment by nikul-serpentcs** [comment]
@hveficent It's Runbot, not Travis.
I Restart runbot build.

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
> > @hveficent It's Runbot, not Travis.
> I Restart runbot build.

Then CI :red_circle:

**Comment by jarroyomorales** [comment]
@nikul-serpentcs Can you attend @jesusVMayor comments? They will fix the CI and we can try to merge it soon, since there haven't been major changes

**Comment by HviorForgeFlow** [comment]
ping @Nikul-Chaudhary @nikul-serpentcs see comments above

**Comment by etobella** [comment]
@nikul-serpentcs Could you attend comments please?

**Review by jarroyomorales — APPROVED** [review]
Nice, all green now :green_apple:!
@jesusVMayor @hveficent @etobella Could you review again?

**Review by HviorForgeFlow — APPROVED** [review]

**Comment by Nikul-Chaudhary** [comment]
now it's ready to merge

**Pull request merged**
